src/overlays/actors/ovl_En_Railgibud/z_en_railgibud.c

```diff
@@ -65,122 +149,999 @@ static DamageTable D_80BA82CC = {
+            Actor_IsActorFacingLink(&this->actor, 0x38E3)) {
+            func_80BA5AF0(this);
+        }
+    }
+}
+
+void func_80BA6440(EnRailgibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, 7);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_DAMAGE);
+    this->unk_3F2 = 0;
+    this->unk_3F4 = 0;
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    this->actor.speedXZ = -2.0f;
+    this->actionFunc = func_80BA64AC;
+}
+
+void func_80BA64AC(EnRailgibud* this, GlobalContext* globalCtx) {
+    if (this->actor.speedXZ < 0.0f) {
+        this->actor.speedXZ += 0.15f;
+    }
+
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        this->unk_405 = -1;
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        if ((this->unk_3F6 > 0) && (this->unk_404 == 0) && (this->unk_3F8 == 0)) {
+            this->actor.hintId = 0x2A;
+            SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06010B88, NULL, this->jointTable, this->morphTable, 26);
+            this->unk_3F8 = 1;
+        }
+        func_80BA6284(this);
+    }
+}
+
+void func_80BA6584(EnRailgibud* this) {
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->unk_3F2 = 10;
+    if (this->unk_3F6 != 0) {
+        func_800BCB70(&this->actor, 0, 0xC8, 0, 0x28);
+    } else {
+        func_800BCB70(&this->actor, 0, 0xC8, 0, 0x28);
+    }
+    this->actionFunc = func_80BA6604;
+}
+
+void func_80BA6604(EnRailgibud* this, GlobalContext* globalCtx) {
+    if (this->actor.colorFilterTimer == 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            func_80BA6664(this);
+        } else {
+            func_80BA6440(this);
+        }
+    }
+
+    if (this->unk_3F2 != 0) {
+        this->unk_3F2--;
+    }
+}
+
+void func_80BA6664(EnRailgibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, 6);
+    this->actor.flags &= ~1;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_DEAD);
+    this->unk_3F2 = 0;
+    this->actionFunc = func_80BA66C8;
+}
+
+void func_80BA66C8(EnRailgibud* this, GlobalContext* globalCtx) {
+    if (this->unk_3F2 > 300) {
+        if (this->actor.shape.shadowAlpha == 0) {
+            if (this->actor.parent != NULL) {
+                Actor_MarkForDeath(&this->actor);
+            } else {
+                this->actor.draw = NULL;
+                this->actor.flags &= ~1;
+                this->actor.update = func_80BA7878;
+            }
+        } else {
+            this->actor.shape.shadowAlpha -= 5;
+            if (this->actor.shape.shadowAlpha < 5) {
+                this->actor.shape.shadowAlpha = 0;
+            }
+        }
+    } else {
+        Math_SmoothStepToS(&this->unk_3E2, 0, 1, 0xFA, 0);
+        Math_SmoothStepToS(&this->unk_3E8, 0, 1, 0xFA, 0);
+        this->unk_3F2++;
+    }
+
+    if ((this->unk_3F2 == 20) && (this->unk_3F6 > 0) && (this->unk_404 == 0) && (this->unk_3F8 == 0)) {
```

Comment:
```suggestion
    if (this->unk_3F2 == 20 && this->unk_3F6 > 0 && this->unk_404 == 0 && this->unk_3F8 == 0) {
```
could probably not use this one if you feel it helps with readability, IMO